boardswarm/src/hifive_p550_storage.rs

```diff
@@ -0,0 +1,209 @@
+    source: &str,
+    dest: &str,
+    mut interval_: time::Interval,
+) -> anyhow::Result<MountInfo> {
+    loop {
+        trace!("Waiting mount point");
+        for mount in MountIter::new()? {
+            let mount = mount?;
+            if mount.source.to_str() == Some(source) && mount.dest.to_str() == Some(dest) {
+                trace!("Found mount point: {:?}", mount);
+                return Ok(mount);
+            }
+        }
+        interval_.tick().await;
+    }
+}
+
+#[derive(Debug)]
+pub struct HifiveP550StorageVolume {
+    path: String,
+    targets: Vec<VolumeTargetInfo>,
+}
+
+impl HifiveP550StorageVolume {
+    fn new(path: String) -> Self {
+        let targets = TARGETS
+            .into_iter()
+            .map(|(name, _)| VolumeTargetInfo {
+                name: name.to_string(),
+                readable: false,
+                writable: true,
+                seekable: false,
+                size: None,
+                blocksize: None,
+            })
+            .collect();
+        Self { path, targets }
+    }
+
+    fn target_path(&self, target: &str) -> Result<String, VolumeError> {
+        let (_, path) = TARGETS
+            .iter()
+            .find(|(name, _)| *name == target)
+            .ok_or(VolumeError::UnknownTargetRequested)?;
+        Ok(format!("{}/{}", self.path, path))
+    }
+}
+
+impl From<std::io::Error> for crate::VolumeError {
+    fn from(_: std::io::Error) -> Self {
+        crate::VolumeError::Internal("Failed to create file".into())
+    }
+}
+
+#[async_trait::async_trait]
+impl Volume for HifiveP550StorageVolume {
+    fn targets(&self) -> (&[VolumeTargetInfo], bool) {
+        (self.targets.as_slice(), true)
+    }
+
+    async fn open(
+        &self,
+        target: &str,
+        _length: Option<u64>,
+    ) -> Result<(VolumeTargetInfo, Box<dyn VolumeTarget>), VolumeError> {
+        if let Some(volume_target) = self.targets.iter().find(|t| t.name == target) {
+            let path = self.target_path(target)?;
+            let file = fs::File::create(path).await.unwrap();
+            Ok((
+                volume_target.clone(),
+                Box::new(HifiveP550StorageTarget { file }),
+            ))
+        } else {
+            warn!("Unknown target requested");
+            Err(VolumeError::UnknownTargetRequested)
+        }
+    }
+
+    async fn commit(&self) -> Result<(), VolumeError> {
+        // TODO: sync filesystem? wait for umounted?
+        Ok(())
+    }
+}
+
+struct HifiveP550StorageTarget {
+    file: fs::File,
+}
+
+impl std::fmt::Debug for HifiveP550StorageTarget {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // TODO make more meaningful
+        f.debug_struct("HifiveP550StorageTarget")
+            .finish_non_exhaustive()
+    }
+}
+
+#[async_trait::async_trait]
+impl VolumeTarget for HifiveP550StorageTarget {
+    async fn write(&mut self, data: Bytes, _offset: u64, completion: crate::WriteCompletion) {
```

Comment:
you can't ignore the offset; either verify the offset or seek to it (and set to `seekable: true`)

Reply:
Done.